frontend/src/Utils/ModelFeatureFlagLoader.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { useSelector } from 'react-redux';
+import { useFeatureFlag } from '@openshift/dynamic-plugin-sdk';
+
+export type ModelFeatureFlagLoaderProps = {
+  flag: string;
+  model: {
+    group: string;
+    version: string;
+    kind: string;
+  };
+};
+
+type K8sState = {
+  k8s: {
+    toJS: () => { RESOURCES: { models: {} } };
+  };
+};
+
+/**
+ * Component to wrap setting and loading of model feature flag.
+ * @param ModelFeatureFlagLoaderProps which model and feature flag to set for given namespace.
+ * @returns null
+ */
+const ModelFeatureFlagLoader: React.FC<ModelFeatureFlagLoaderProps> = ({ flag, model }) => {
+  const [, setFlag] = useFeatureFlag(flag);
+  const isPresent = useSelector<K8sState, boolean>(({ k8s }) => {
+    const { RESOURCES } = k8s?.toJS();
+    return !!Object.values(RESOURCES.models).find(({ apiGroup, apiVersion, kind }) => {
+      return apiGroup === model.group && apiVersion === model.version && kind === model.kind;
+    });
+  });
```

Comment:
Can you use `useK8sModel` instead of direct access to redux?

Reply:
Unfortunately that hook is not exported. I'll open a PR in SDK to export it.